### PR TITLE
Add scripts for downloading data from polish court API

### PR DESCRIPTION
**Pull request by binkjakub**

### CODE REVIEW

File: juddges/data/pl_court_api.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class PolishCourtAPI:
+    def __init__(self) -> None:
+        self.url = "https://apiorzeczenia.wroclaw.sa.gov.pl/ncourt-api"
+
+    def get_number_of_judgements(self, params: dict[str, Any] | None = None) -> int:
+        if params is None:
+            params = {}
+
+        params |= {"limit": 0}
```
Comment (asawczyn) [inline juddges/data/pl_court_api.py:18]: Do you want to overwrite "limit" ?

File: scripts/download_pl_content.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        try:
+            collection.bulk_write(data_batch)
+        except BulkWriteError as err:
+            logger.error(err)
+
+    @retry(
+        wait=wait_random_exponential(multiplier=1, max=60),
+        retry=retry_if_exception_type(HTTPError),
+        stop=stop_after_attempt(3),
```
Comment (asawczyn) [inline scripts/download_pl_content.py]: is it sufficient?

File: scripts/download_pl_metadata.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+class MetadataDownloader:
+    def __init__(self, mongo_uri: str, batch_size: int):
+        self.mongo_uri = mongo_uri
+        self.batch_size = batch_size
+
+    @retry(
+        wait=wait_random_exponential(multiplier=1, max=60),
+        retry=retry_if_exception_type(HTTPError),
+        stop=stop_after_attempt(3),
```
Comment (asawczyn) [inline scripts/download_pl_metadata.py]: as above

**Review by asawczyn — CHANGES_REQUESTED** [review]

File: juddges/data/pl_court_api.py (hunk 1 above)
Comment (binkjakub) [reply juddges/data/pl_court_api.py:18]: Yes, I need to get number of judgements but without fetching data of any judgement, so `limit` must be 0

Comment (asawczyn) [reply juddges/data/pl_court_api.py:18]: But someone can pass the limit param, and you don't inform/warn them about overwriting.

Comment (asawczyn) [reply juddges/data/pl_court_api.py:18]: Imo, just add assert.

Comment (binkjakub) [reply juddges/data/pl_court_api.py:18]: No, that's fully correct. By construction this limit must be 0, and passing it won't affect results (returned result is always the same regardless of limit, that's how this API is built)

Comment (binkjakub) [reply juddges/data/pl_court_api.py:18]: I'll add warning

File: scripts/download_pl_content.py (hunk 2 above)
Comment (binkjakub) [reply scripts/download_pl_content.py]: increased to 5, and changed min/max times

File: scripts/download_pl_metadata.py (hunk 3 above)
Comment (binkjakub) [reply scripts/download_pl_metadata.py]: increased to 5, and changed min/max times

**Review by asawczyn — APPROVED** [review]

**Pull request merged**